### ISSUE TITLE
Could not open wss://localhost:44382/_framework/debug/ws-proxy

### ISSUE THREAD

**Issue by ylr-research** [dotnet/aspnetcore#35221]
### Background

I have a Blazor app with ASP.NET Core Hosted. I added a Background service for long running task and from this service I call a Hub to push a notification to the clients once the task is done. When I run the project from VS 2022 latest preview I get the below error. Basically the app doesn't finish loading. When I click Ok then it shutdown the app.

![image](https://user-images.githubusercontent.com/63487874/128872839-24ce1ba2-1f66-4939-8bfc-81ee177be810.png)

### Visual Studio Output

\Workflow\\SourceDataLoad\\XFWebSourceDataLoadComponent.razor","sources":[],"checksums":[]},"breakpoints":[{"line":253,"column":13}],"lines":[253]},"seq":4}},"level":0}
{"tag":"dap.send","timestamp":1628597988935,"metadata":{"connectionId":0,"message":{"seq":6,"type":"response","request_seq":4,"command":"setBreakpoints","success":true,"body":{"breakpoints":[{"id":1,"verified":false,"message":"Unbound breakpoint"}]}}},"level":0}
{"tag":"dap.receive","timestamp":1628597989136,"metadata":{"connectionId":0,"message":{"type":"request","command":"setExceptionBreakpoints","arguments":{"filters":[]},"seq":5}},"level":0}
{"tag":"dap.send","timestamp":1628597989137,"metadata":{"connectionId":0,"message":{"seq":7,"type":"response","request_seq":5,"command":"setExceptionBreakpoints","success":true,"body":{}}},"level":0}
{"tag":"dap.receive","timestamp":1628597989172,"metadata":{"connectionId":0,"message":{"type":"request","command":"configurationDone","arguments":{},"seq":6}},"level":0}
{"tag":"dap.send","timestamp":1628597989172,"metadata":{"connectionId":0,"message":{"seq":8,"type":"response","request_seq":6,"command":"configurationDone","success":true,"body":{}}},"level":0}
{"tag":"runtime.launch","timestamp":1628597990027,"message":"Discovered target URL from /json/version","metadata":{"url":"ws://localhost/devtools/browser/8988f730-358a-4393-b7c6-6bd83c18cc79","fixed":"ws://localhost:56593/devtools/browser/8988f730-358a-4393-b7c6-6bd83c18cc79"},"level":0}
{"tag":"dap.send","timestamp":1628597991401,"metadata":{"connectionId":0,"message":{"seq":9,"type":"event","event":"output","body":{"category":"telemetry","output":"js-debug/dap/operation","data":{"errors":[],"initialize":{"operation":"initialize","totalTime":12.2,"max":12.2,"avg":12.2,"stddev":null,"count":1,"failed":0},"!initialize.errors":[],"initialize.errors":[],"setBreakpoints":{"operation":"setBreakpoints","totalTime":1.6,"max":1.6,"avg":1.6,"stddev":null,"count":1,"failed":0},"!setBreakpoints.errors":[],"setBreakpoints.errors":[],"setExceptionBreakpoints":{"operation":"setExceptionBreakpoints","totalTime":0.9,"max":0.9,"avg":0.9,"stddev":null,"count":1,"failed":0},"!setExceptionBreakpoints.errors":[],"setExceptionBreakpoints.errors":[],"configurationDone":{"operation":"configurationDone","totalTime":0.6,"max":0.6,"avg":0.6,"stddev":null,"count":1,"failed":0},"!configurationDone.errors":[],"configurationDone.errors":[]}}}},"level":0}
{"tag":"runtime.launch","timestamp":1628598046820,"message":"Launch returned error","metadata":{"error":{"message":"Unable to launch browser: \"Could not open wss://localhost:44382/_framework/debug/ws-proxy?browser=ws%3A%2F%2Flocalhost%3A56593%2Fdevtools%2Fbrowser%2F8988f730-358a-4393-b7c6-6bd83c18cc79\"","stack":"Error: Unable to launch browser: \"Could not open wss://localhost:44382/_framework/debug/ws-proxy?browser=ws%3A%2F%2Flocalhost%3A56593%2Fdevtools%2Fbrowser%2F8988f730-358a-4393-b7c6-6bd83c18cc79\"\n    at P.prepareLaunch (c:\\program files\\microsoft visual studio\\2022\\preview\\common7\\ide\\commonextensions\\microsoft\\jsdiagnostics\\debugger\\debugAdapter\\out\\src\\targets\\browser\\browserLauncher.js:115:19)"},"wasCancelled":true,"name":"P"},"level":2}
{"tag":"dap.send","timestamp":1628598046820,"metadata":{"connectionId":0,"message":{"seq":10,"type":"response","request_seq":2,"command":"launch","success":false,"body":{"error":{"id":9240,"format":"Unable to launch browser: \"Could not open wss://localhost:44382/_framework/debug/ws-proxy?browser=ws%3A%2F%2Flocalhost%3A56593%2Fdevtools%2Fbrowser%2F8988f730-358a-4393-b7c6-6bd83c18cc79\"","showUser":false}}}},"level":0}
{"tag":"dap.send","timestamp":1628598048871,"metadata":{"connectionId":0,"message":{"seq":11,"type":"event","event":"output","body":{"category":"telemetry","output":"js-debug/error","data":{"!error":{"error":{"message":"read ECONNRESET","name":"Error","stack":"Error: read ECONNRESET\n    at TCP.onStreamRead (internal/stream_base_commons.js:209:20)"}},"error":{"error":{"message":"read ECONNRESET","name":"Error","stack":"Error: read ECONNRESET\n    at TCP.onStreamRead (internal/stream_base_commons.js:209:20)"}},"exceptionType":"uncaughtException"}}}},"level":0}
{"tag":"dap.send","timestamp":1628598048871,"message":"Message not sent. Connection was closed.","level":2}
{"tag":"runtime.exception","timestamp":1628598048871,"message":"Unhandled error in debug adapter","metadata":{"message":"read ECONNRESET","stack":"Error: read ECONNRESET\n    at TCP.onStreamRead (internal/stream_base_commons.js:209:20)"},"level":3}
{"tag":"dap.send","timestamp":1628598051835,"metadata":{"connectionId":0,"message":{"seq":12,"type":"event","event":"output","body":{"category":"telemetry","output":"js-debug/dap/operation","data":{"errors":[],"launch":{"operation":"launch","totalTime":60024.5,"max":60024.5,"avg":60024.5,"stddev":null,"count":1,"failed":1},"!launch.errors":[{"error":{"message":"Unable to launch browser: \"Could not open wss://localhost:44382/_framework/debug/ws-proxy?browser=ws%3A%2F%2Flocalhost%3A56593%2Fdevtools%2Fbrowser%2F8988f730-358a-4393-b7c6-6bd83c18cc79\"","name":"Error","stack":"Error: Unable to launch browser: \"Could not open wslocalhost:44382ws-proxy?browser=ws%3A%2F%2Flocalhost%3A56593%2Fdevtools%2Fbrowser%2F8988f730-358a-4393-b7c6-6bd83c18cc79\"\n    at P.prepareLaunch (c:browserLauncher.js:115:19)"}}],"launch.errors":[{"error":{"message":"Unable to launch browser: \"Could not open wss://localhost:44382/_framework/debug/ws-proxy?browser=ws%3A%2F%2Flocalhost%3A56593%2Fdevtools%2Fbrowser%2F8988f730-358a-4393-b7c6-6bd83c18cc79\"","name":"Error","stack":"Error: Unable to launch browser: \"Could not open wslocalhost:44382ws-proxy?browser=ws%3A%2F%2Flocalhost%3A56593%2Fdevtools%2Fbrowser%2F8988f730-358a-4393-b7c6-6bd83c18cc79\"\n    at P.prepareLaunch (c:browserLauncher.js:115:19)"}}]}}}},"level":0}
{"tag":"dap.send","timestamp":1628598051835,"message":"Message not sent. Connection was closed.","level":2}

### Startup.cs

![image](https://user-images.githubusercontent.com/63487874/128874193-4718d704-6dc7-4b21-a16d-99bdb9050905.png)

### Background Service

```
    public class MyBackgroundService : BackgroundService
    {
        private readonly IHubContext<MyHub, IMyHub> _hubContext;

        public StageProcessingBackgroundService(IHubContext<MyHub, IMyHub> hubContext)
        {
            _hubContext = hubContext;
        }

        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
        {
            while (!stoppingToken.IsCancellationRequested)
            {
                   //await some IO operation
            }
        }
    }
```

**Comment by mkArtakMSFT** [comment]
@lewing can you please look into this? Thanks!

**Comment by ylr-research** [comment]
Thank you for the quick response. I was able to get it to work. Here is the outcome of my troubleshooting.

According to the [documentation](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services?view=aspnetcore-5.0&tabs=visual-studio), the `StartAsync` method should run before the app pipeline, and such default behavior can be changed by having the hosted service registered from at the `Program.cs`, which is how I actually got it to work. Apparently having the hosted service registered at the `ConfigureServices` method from `Startup.cs` does matter and causes the app to crash. I would just recommend to add to the documentation a clear example on how the default behavior should be setup.

This issue can be closed. Thanks you all for the help.
